src/main/java/build/buildfarm/instance/stub/StubInstance.java

```diff
@@ -73,10 +73,21 @@
   private final Channel channel;
   private final ByteStreamUploader uploader;
 
+  private static final InputStream nullInputStream = new InputStream() {
+    @Override
+    public int read() {
```

Comment:
I'm not sure if you're worried about efficiency here, but you might want to override some of the other InputStream methods as well, e.g., the other read() method.

Reply:
Agreed, if I were leaving this in.  Throwing it out per the awkward advance() fix below.